### ISSUE TITLE
Not warning about custom hooks as properties

### ISSUE THREAD

**Issue by esamattis** [Gelio/tslint-react-hooks#14]
This should warn

```ts
if (cond) {
    MyHooks.useHook();
}
```

like this does

```ts
if (cond) {
    React.useEffect();
}
```

**Comment by CruseCtrl** [comment]
This is a [deliberate decision](https://github.com/facebook/react/blob/c11015ff4f610ac2924d1fc6d569a17657a404fd/packages/eslint-plugin-react-hooks/src/RulesOfHooks.js#L34) from `eslint-plugin-react-hooks` to prevent false positives

If you import the function and use it directly then it will work
```typescript
import { useHook } from 'MyHooks';

if (cond) {
  useHook(); // This will have a warning
}
```

And if you check [the docs](https://reactjs.org/docs/hooks-effect.html), they tend to do this with the default hooks too, rather than using e.g. `React.useEffect()`

**Comment by Gelio** [comment]
@CruseCtrl Thank you for the explanation.

When implementing this rule, I strictly followed each heuristic that was made in `eslint-plugin-react-hooks`. Detecting using namespaced hooks is one of them.

If the eslint plugin changes this logic, I will try to update this project accordingly 🙂 For now, I am closing this issue to keep the behavior as is, matching the original rule
